client/blocks/domain-to-plan-nudge/index.jsx

```diff
@@ -0,0 +1,333 @@
+import React, { PropTypes, Component } from 'react';
+import { localize } from 'i18n-calypso';
+import { connect } from 'react-redux';
+import { get } from 'lodash';
+import page from 'page';
+
+/**
+ * Internal dependencies
+ */
+import Button from 'components/button';
+import DismissibleCard from 'blocks/dismissible-card';
+import { recordTracksEvent } from 'state/analytics/actions';
+import { getSelectedSiteId } from 'state/ui/selectors';
+import { getSite, isCurrentSitePlan } from 'state/sites/selectors';
+import { plansList, PLAN_FREE, PLAN_PERSONAL } from 'lib/plans/constants';
+import { storedCardPayment } from 'lib/store-transactions';
+import { getStoredCards } from 'state/stored-cards/selectors';
+import QueryStoredCards from 'components/data/query-stored-cards';
+import { emptyCart } from 'lib/cart-values';
+import { add } from 'lib/cart-values/cart-items';
+import { submitTransaction } from 'lib/upgrades/actions/checkout';
+import PlanPrice from 'my-sites/plan-price';
+import PlanIcon from 'components/plans/plan-icon';
+import Gridicon from 'components/gridicon';
+import { errorNotice, infoNotice, removeNotice } from 'state/notices/actions';
+import { getCurrentUserCurrencyCode } from 'state/current-user/selectors';
+import { recordOrder } from 'lib/analytics/ad-tracking';
+import { getPlanRawPrice } from 'state/plans/selectors';
+import {
+	getPlanDiscountedRawPrice,
+	getPlanRawDiscount,
+	getPlansBySiteId
+} from 'state/sites/plans/selectors';
+import QuerySitePlans from 'components/data/query-site-plans';
+import formatCurrency from 'lib/format-currency';
+
+const debug = debugFactory( 'calypso:domain-to-plan-nudge' );
+
+class DomainToPlanNudge extends Component {
+
+	constructor() {
+		super( ...arguments );
+		this.oneClickUpgrade = this.oneClickUpgrade.bind( this );
+		this.handleTransactionComplete = this.handleTransactionComplete.bind( this );
+		this.state = {
+			isSubmitting: false,
+			noticeId: null
+		};
+	}
+
+	static propTypes = {
+		hasFreePlan: PropTypes.bool,
+		site: PropTypes.object,
+		siteId: PropTypes.number,
+		translate: PropTypes.func
+	};
+
+	isVisible() {
+		const {
+			site,
+			hasFreePlan,
+			sitePlans,
+			storedCard,
+			rawPrice
+		} = this.props;
+
+		return sitePlans.hasLoadedFromServer &&
+			storedCard &&
+			rawPrice &&
+			site &&           //site exists
+			site.wpcom_url && //has a mapped domain
+			hasFreePlan;      //has a free wpcom plan
+	}
+
+	getCartItem() {
+		const {
+			productSlug,
+			productId
+		} = this.props;
+
+		return {
+			product_id: productId,
+			product_slug: productSlug,
+			free_trial: false,
+			is_domain_registration: false
+		};
+	}
+
+	handleTransactionComplete( error, data ) {
+		const { siteId, translate, userCurrency, rawPrice, discountedRawPrice } = this.props;
+
+		const price = discountedRawPrice || rawPrice;
+
+		debug( 'transaction complete', error, data );
+		// see transaction-steps-mixin for matching checkout analytics
```

Comment:
Nit: if you add "TODO: ..." before the text, it will be highlighted in PHPStorm so we don't skip it thinking it's just a comment for further clarification of code. :)